### DISCUSSION TITLE
_toConsumableArray throw type error for empty object.

### DISCUSSION THREAD

**Discussion by PareshRadadiya** [babel/babel#12052] [Q&A] [answered]
The `_toConsumableArray` helper in babel 7 throw type error if we pass an empty object. It used to work fine in the previous babel 6 version. Is it a bug or intentional change?

**Comment by JLHwung** [comment, marked as answer]
This is expected. Including empty object, a plain object is not iterable and thus `_toConsumableArray` will throw.
```js
for (const v of {});
// throws {} is not iterable
```
